packages/ruleset/src/functions/api-symmetry.js

```diff
@@ -0,0 +1,368 @@
+    // of the canonical schema.
+    ['Summary', 'Prototype', 'Patch'].forEach(variantType => {
+      const variantSchemaName = `${canonicalSchemaName}${variantType}`;
+      const variantSchema = apidef.components.schemas[`${variantSchemaName}`];
+      if (variantSchema && isObjectSchema(variantSchema)) {
+        logger.info(
+          `${ruleId}: Checking variant schema ${variantSchemaName} against canonical schema:`
+        );
+        if (
+          !checkForGraphFragmentPattern(
+            variantSchema,
+            canonicalSchema,
+            canonicalSchemaPath.split('.'),
+            variantType === 'Prototype',
+            {
+              refMap: pathToReferencesMap,
+              allSchemas: apidef.components.schemas,
+            }
+          )
+        ) {
+          logger.debug(
+            `${ruleId}: determined that ${variantSchemaName} is not a proper graph fragment of ${canonicalSchemaName}`
+          );
+          errors.push({
+            message:
+              'Variant schema should be a graph fragment of the canonical schema',
+            path: ['components', 'schemas', variantSchemaName],
+          });
+        }
+      }
+    });
+  }
+
+  return errors;
+}
+
+/**
+ * Determine if the variant schema is indeed a proper graph fragment of the
+ * canonical schema, using the following conditions:
+ * - The variant does not define additional or pattern properties
+ * - The variant does not define any properties that do not exist on the
+ *   canonical schema (exists = present and has same type)
+ * - The variant omits at least one property from the canonical schema
+ *
+ * @param {object} variant - the variant schema to check
+ * @param {object} canonical - the canonical schema to check against
+ * @param {array} canonicalPath - array of JSON path segments leading to canonical schema
+ * @param {boolean} considerWriteOnly - whether or not to take writeOnly into account
+ * @param {object} schemaFinder - an object holding data to help find canonical schema versions
+ *                    refMap - map of paths to schema references
+ *                    allSchemas - all of the schemas in components
+ * @returns {boolean} true if the variant is a proper graph fragment
+ */
+function checkForGraphFragmentPattern(
+  variant,
+  canonical,
+  canonicalPath,
+  considerWriteOnly,
+  schemaFinder
+) {
+  let variantOmitsProperties = false;
+  let isValidGraphFragment = true;
+
+  // The function is wrapped in a partial so that we can track `variantOmitsProperties`
+  // across the entire schema without overriding it during recursive invocations of the
+  // graph fragment checker.
+
+  function isGraphFragment(variant, canonical, canonicalPath, fromApplicator) {
+    // A variant schema cannot allow extraneous properties and still be
+    // an explicit graph fragment of the canonical schema.
+    if (variant.additionalProperties) {
+      logger.info(
```

Comment:
I use `info` level messages here to be able to tell the user all the reasons why the schema isn't a graph fragment without trying to a) include all the reasons in the primary message or b) making the user sift through the debug logs to see the reasons. I'm including a note in the docs that info level logs will give the user more specific info for this rule